packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -273,24 +274,26 @@ export function createRootHooks<
     const shouldAbortOnUnmount =
       opts?.trpc?.abortOnUnmount ?? config?.abortOnUnmount ?? abortOnUnmount;
 
-    const hook = __useQuery({
-      ...ssrOpts,
-      queryKey: queryKey as any,
-      queryFn: (queryFunctionContext) => {
-        const actualOpts = {
-          ...ssrOpts,
-          trpc: {
-            ...ssrOpts?.trpc,
-            ...(shouldAbortOnUnmount
-              ? { signal: queryFunctionContext.signal }
-              : {}),
-          },
-        };
+    const hook = __useQuery(
+      {
+        ...ssrOpts,
+        queryKey: queryKey as any,
+        queryFn: (queryFunctionContext) => {
+          const actualOpts = {
+            ...ssrOpts,
+            trpc: {
+              ...ssrOpts?.trpc,
+              ...(shouldAbortOnUnmount
+                ? { signal: queryFunctionContext.signal }
+                : {}),
+            },
+          };
 
-        return client.query(...getClientArgs(queryKey, actualOpts));
+          return client.query(...getClientArgs(queryKey, actualOpts));
+        },
       },
-      context: ReactQueryContext,
-    }) as UseTRPCQueryResult<unknown, TError>;
+      queryClient,
```

Comment:
https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#removed-custom-context-prop-in-favor-of-custom-queryclient-instance